generated/server/router.go

```diff
@@ -1,18 +1,37 @@
 package server
 
 import (
+	"errors"
 	"net/http"
+	"time"
 
 	"github.com/gorilla/mux"
 
 	gContext "github.com/gorilla/context"
 	"golang.org/x/net/context"
+
+	"gopkg.in/tylerb/graceful.v1"
 )
 
 type contextKey struct{}
 
-func SetupServer(r *mux.Router, c Controller) http.Handler {
+type Server struct {
+	Handler http.Handler
+	Port    int
```

Comment:
the reason i made `Handler` public is that I needed it for my test: `httptest.NewServer(s.Handler)`. I imagine others are going to want to do something similar. Maybe I should expose a `ServeForTesting` method instead? Any other ideas?

will make port private